crates/jxl-image/src/lib.rs

```diff
@@ -231,7 +231,12 @@ impl<Ctx> Bundle<Ctx> for ExtraChannelInfo {
 
         let ty_id = read_bits!(bitstream, Enum(ExtraChannelTypeRaw))?;
         let bit_depth = BitDepth::parse(bitstream, ())?;
+
         let dim_shift = read_bits!(bitstream, U32(0, 3, 4, 1 + u(3)))?;
+        if (1usize << dim_shift) > 8 {
```

Comment:
As discussed in jxl discord chat, there's something wrong in the check in libjxl; this makes U32 value of 4 (`10` in bitstream) instantly invalid. Instead we could defer the check to the frame header decoding.

Reply:
Agreed, it seems to be an early check for `dim_shift` when `ec_upsampling` is not yet known.